### PR TITLE
add `plugin-paraglide-js` [closes #1475]

### LINKED ISSUE #1475: paraglide ide extension plugin
## Problem 

No paraglide plugin exists for inlang yet. Thus, the ide extension app doesn't work. 

## Proposal 

Release a paraglide plugin for inlang that makes the ide extension work. 

- Only the `messageReferenceMatchers` need to be defined. 
- Messages are always referenced via `m.*` 

### Pseudocode

```
1. Check if the code file contains `"@inlang/paraglide-js/*/messages" to avoid false positives. 
2. Search the code for `m.*()` references.

### PR DESCRIPTION
**Pull request by felixhaeberle**
This plugin provides the integration for `paraglide-js` into the inlang VSCode extension. It allows you to extract messages from your code, view them inline in the text editor and get lints for your messages.

### CODE REVIEW

**Comment by felixhaeberle** [comment]
fixes #1475

**Comment by felixhaeberle** [comment]
Ui that was a bit too fast. The code which should be reviewed is in /paraglide-js/paraglide-js-plugin

File: inlang/source-code/paraglide/paraglide-js-plugin/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+	"name": "@inlang/plugin-paraglide-js",
+	"version": "0.0.0",
```
Comment (samuelstroschein) [inline inlang/source-code/paraglide/paraglide-js-plugin/package.json:3]: version 1 or?

File: inlang/source-code/paraglide/paraglide-js-plugin/src/ideExtension/messageReferenceMatchers.test.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
+import { describe, expect, it } from "vitest"
+import { parse } from "./messageReferenceMatchers.js" // Replace with the actual filename
+
+describe("Paraglide Message Parser", () => {
+	it("should match simple m function call", () => {
+		const sourceCode = `
+		import * as m from "@inlang/paraglide-js/example/messages";
```
Comment (samuelstroschein) [inline inlang/source-code/paraglide/paraglide-js-plugin/src/ideExtension/messageReferenceMatchers.test.ts:7]: please add a test for different namespaces. 

```ts
import * as m from "@inlang/paraglide-js/{namespace}/messages"
```

as well as no namespace (which I am going to allow to simplify the getting started https://github.com/inlang/monorepo/discussions/1464#discussioncomment-7280355)

```ts
import * as m from "@inlang/paraglide-js/messages"
```

**Review by samuelstroschein — CHANGES_REQUESTED** [review]

File: inlang/source-code/paraglide/paraglide-js-plugin/package.json (hunk 1 above)
Comment (felixhaeberle) [reply inlang/source-code/paraglide/paraglide-js-plugin/package.json:3]: already created a changeset

File: inlang/source-code/paraglide/paraglide-js-plugin/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Plugin } from "@inlang/plugin"
+import { displayName, description } from "../marketplace-manifest.json"
+import { PluginSettings } from "./settings.js"
+import { ideExtensionConfig } from "./ideExtension/config.js"
+
+const id = "plugin.inlang.plugin-paraglide-vscode"
```
[the remark this answers is not in the document]
Comment (felixhaeberle) [reply inlang/source-code/paraglide/paraglide-js-plugin/src/plugin.ts:6]: Solved 👍

File: inlang/source-code/paraglide/paraglide-js-plugin/src/ideExtension/messageReferenceMatchers.test.ts (hunk 2 above)
Comment (felixhaeberle) [reply inlang/source-code/paraglide/paraglide-js-plugin/src/ideExtension/messageReferenceMatchers.test.ts:7]: Solved 👍

**Review by samuelstroschein — APPROVED** [review]

**Comment by felixhaeberle** [comment]
@samuelstroschein This can be merged. 

--- 
@floriandwt How can it be added to the marketplace? Does it work automatically?

**Pull request merged**